rust/rope/src/engine.rs

```diff
@@ -39,6 +39,9 @@ pub struct Engine {
 #[derive(Debug)]
 struct Revision {
     rev_id: usize,
+    /// The largest undo group number of any edit in the history up to this
```

Comment:
For now that approach to allocating numbers is compatible with this approach, and if anyone ever wants to optimize memory and we can guarantee that revisions are allocated in that way, removing this field is an easy optimization.